iperf2: Fix compilation failure with --disable-ipv6

terasawa-lin opened the pull request:
# Purpose of pull request
This PR fixes the build error of iperf2 when ipv6 is disabled.

We can reproduce the error by adding the following config to local.conf.
`DISTRO_FEATURES_remove = "ipv6"`

This problem is reported in iperf2 upstream as below.
https://sourceforge.net/p/iperf2/tickets/48/

We backport the corresponding patch from the upstream.

hiraku-wfs commented on recipes-debian/iperf2/iperf2/0003-fix_build_without_ipv6.patch:
```diff
@@ -0,0 +1,21 @@
+--- a/src/Listener.cpp
```
Please use the patch which was really merged to the iperf2 upstream. You can use git-format-patch.

masami256 replied:
It looks this patch back ported from "[fix compile when --disable-ipv6 is configured. Thanks to Gaetano Catalli](https://sourceforge.net/p/iperf2/code/ci/e7dc021b7f791256ef280bb25a398c03e4433c2a/)" patch. If so, please describe where patch come from.

terasawa-lin committed:
iperf2: Fix Compilation fails with --disable-ipv6

Add a patch for this commit
https://sourceforge.net/p/iperf2/code/ci/e7dc021b7f791256ef280bb25a398c03e4433c2a/

hiraku-wfs commented:
Thanks!
I modified the title and description of this PR to make it easy to understand.
